engine/src/test/java/io/camunda/zeebe/engine/processing/resource/ResourceDeletionTest.java

```diff
@@ -61,6 +70,119 @@ public void shouldWriteDeletedEventsForMultipleDecisions() {
     verifyResourceIsDeleted(drgKey);
   }
 
+  @Test
+  public void shouldCreateIncidentIfOnlyDecisionVersionIsDeleted() {
+    // given
+    final long drgKey = deployDrg(DRG_SINGLE_DECISION);
+    final var processId = deployProcessWithBusinessRuleTask("jedi_or_sith");
+
+    // when
+    engine.resourceDeletion().withResourceKey(drgKey).delete();
```

Comment:
👍 This is a really nice and clear commit, great job!

These tests cover deletion of the DRG.
❓ Should it also possible to delete decisions using the decision key directly? I'm not sure what should happen to related DRG, but for the decision it would be clear. If it should be possible, let's add some test cases for this.

Reply:
It will not be possible to delete decisions directly. It's similar to a deployment, you deploy the DRG and we create the decisions. You delete the DRG and we delete the decisions.